monix-catnap/src/main/scala/com/avast/sst/monix/catnap/CircuitBreakerModule.scala

```diff
@@ -0,0 +1,55 @@
+package com.avast.sst.monix.catnap
+
+import cats.effect.{Clock, Sync}
+import cats.syntax.flatMap._
+import monix.catnap.CircuitBreaker
+import org.slf4j.LoggerFactory
+
+class CircuitBreakerModule[F[_]](implicit F: Sync[F]) {
+
+  private lazy val logger = LoggerFactory.getLogger("CircuitBreaker")
```

Comment:
why are you providing custom logger name? isn't the standard to use class as logger name?

Reply:
also, why don't you provide some logger that works well with cats-effect? seems kinda annoying too always having to wrap it yourself

Reply:
it might also be useful to add `name` into the logger name so that logback can be configured to enable tracing for some problematic upstreams - the name then would be for example `CircuitBreaker.rules-storage`

Reply:
> logger that works well with cats-effect

good point! But I must say, that I don't know, if it's a solved problem -- or solved well, to be precise. I know there exist things like `log4cats` or `log-effect`, but I have no idea about their qualities. (I've made a small overview of various logging libs for Scala a while ago, but that's on our internal Scala guild wiki.)

> tracing for some problematic upstreams

:+1:

Reply:
well at the very least i think you can write a custom wrapper around `Logger`interface so that the users don't have to wrap those things everywhere (might be different PR) ... we also do some support for context propagation during logging but that might be too specific to implement here

Reply:
The "custom" name is a leftover after refactorings. Of course it should be the class name. The reason I use `String` instead of `Class` is that whenever it's part of `object` it looks nasty (dollar sign at the end) so I tried to use the plain version without dollar sign (this does not apply to this case because this is inside `class` but as I said it's some leftover after refactorings).

I don't think the upstream name should be part of the logger name. I do write out the upstream name in the log message though so I think that is sufficient.

And finally the topic about logging library. Of course I was looking for some `F-compatible` logging library but it seems to me there is none that would be mature and spread enough to consider it for this library. Adding a dependency is a burden and I am not ready to do that with a logging library just now. That being sad I only added `slf4j-api` (the lowest common denominator hopefully everyone is ok with) and I use that internally - which is not ergonomic but who cares, it's just a few logging lines. However that does not mean everyone else has to do that. Users of this library can of course choose any logging library they want. This way I am not forcing anyone to use my chosen library but I am also able to log messages inside the library and it's almost dependency-free (`slf4j-api` seems reasonable to me). Makes sense?

Reply:
> I don't think the upstream name should be part of the logger name. I do write out the upstream name in the log message though so I think that is sufficient.

Having the name part of logger name is strictly better because you can then control the logging using `logback.xml`

about logging library, it would make sense to have a FP-ready slf4j api wrapper readily available to for people but i'm not gonna force you to write one

Reply:
Well, I don't want to implement yet another logging framework. I believe that one of those libraries will catch up and we'll use it but for now anyone can choose whatever they want and I am not forcing anything on them.

Reply:
i still think logger name should contain name for the logger

Reply:
It does: https://github.com/avast/scala-server-toolkit/pull/56/files#diff-184610edf024609b740a39c74e1e9b6bR38